test/gpu/vllm/vllm_test.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2024 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package vllm
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/mount"
+	"gvisor.dev/gvisor/pkg/test/dockerutil"
+	"gvisor.dev/gvisor/test/benchmarks/harness"
+)
+
+func TestMain(m *testing.M) {
+	harness.Init()
+	os.Exit(m.Run())
+}
+
+// BenchmarkVLLM runs a vLLM workload.
+func BenchmarkVLLM(b *testing.B) {
+	doVLLMTest(b)
+}
+
+func doVLLMTest(b *testing.B) {
+	serverMachine, err := harness.GetMachine()
+	if err != nil {
+		b.Fatalf("failed to get machine: %v", err)
+	}
+	defer serverMachine.CleanUp()
+
+	b.Run("opt-125", func(b *testing.B) {
+		ctx := context.Background()
+
+		b.ResetTimer()
+		b.StopTimer()
+
+		serverCtr := serverMachine.GetContainer(ctx, b)
+		defer serverCtr.CleanUp(ctx)
+		if err := harness.DropCaches(serverMachine); err != nil {
+			b.Skipf("failed to drop caches: %v. You probably need root.", err)
+		}
+
+		// Run vllm.
+		runOpts := dockerutil.GPURunOpts()
+		runOpts.Image = "gpu/vllm"
+		runOpts.Env = []string{"PYTHONPATH=$PYTHONPATH:/vllm"}
+
+		if err := serverCtr.Spawn(ctx, runOpts); err != nil {
+			b.Errorf("failed to run container: %v", err)
+		}
+		if out, err := serverCtr.WaitForOutput(ctx, "Uvicorn running on http://0.0.0.0:8000", 10*time.Minute); err != nil {
+			b.Fatalf("failed to start vllm model: %v %s", err, out)
+		}
+
+		for i := 1; i <= b.N; i++ {
```

Comment:
This should start at `i := 0`. (But see comment below as to why maybe the loop shouldn't exist anyway)